Flow.Rmd

```diff
@@ -123,13 +169,407 @@ gating(gt, gs)
 
 Now that the workflow is finished, here is a look at all cell types identified:
 
-```{r plot-cell-types, fig.height=8, fig.width=8}
+```{r flow-plot-cell-types, fig.height=8, fig.width=8}
 autoplot(gs[[1]], strip.text = "gate", bins=100, merge=F, axis_inverse_trans=F) + flow_theme
 ```
 
 This should then be comparable to what was in the [OMIP-021 publication](https://onlinelibrary.wiley.com/doi/full/10.1002/cyto.a.22475), and here are relevant figures demonstrating the similarity of the cell subsets captured:
 
-```{r OMIP-021-figure, fig.align='center', fig.cap='Gating strategy illustrated with annotated figures from OMIP-021', echo=FALSE}
-knitr::include_graphics('images/OMIP021.png')
+```{r flow-OMIP-021-figure, results='asis', fig.align='center', fig.cap='Gating strategy illustrated with annotated figures from OMIP-021', echo=FALSE, out.width = '100%'}
+knitr::include_graphics('images/flow-OMIP021-populations.png')
+```
+
+## Advanced Analysis Automation (OMIP-030)
+
+Advanced Example: [OMIP-030: Characterization of eight major human CD4+ T helper subsets, including Tregs, via surface markers (FR-FCM-ZZWU)](https://flowrepository.org/id/FR-FCM-ZZWU)
+
+### Gating
+
+Load a flow workspace, which will contain all gaiting and cell expression information:
+
+```{r flow-load-workspace}
+data_dir <- dir_create('data/flow_files/OMIP-030')
+zip_url <- 'https://storage.googleapis.com/t-cell-data/flow_files/OMIP-030.zip'
+zip_file <- fs::path(data_dir, 'OMIP-030.zip')
+if (!file_exists(zip_file)) {
+  curl_download(zip_url, zip_file) %>% unzip(exdir=data_dir)
+}
+ws <- openWorkspace(fs::path(data_dir, 'OMIP-030.wsp'))
+gs <- parseWorkspace(ws, name='TCells', path=data_dir, isNcdf=FALSE)
+
+# Extract single GatingHierarchy from GatingSet
+gh <- gs[[1]]
 ```
 
+Show the correspondence between the flow cytometer channels and the marker names in the panel.
+
+```{r flow-show-markers, results='markup'}
+getData(gh)@parameters@data %>% select(name, desc)
+```
+
+Plot the gating workflow starting from where T Cells are first identified:
+
+```{r flow-show-workflow, fig.width=14, fig.height=8}
+plot(gs, 'T Cells', width=6, height=1, fontsize=18, shape='rectangle')
+```
+
+
+TODO: Explain gating via differentiation model
+
+```{r flow-OMIP-030-figure, results='asis', fig.align='center', fig.cap='Gating strategy and cell differentiation overlay from OMIP-030', echo=FALSE}
+knitr::include_graphics('images/flow-OMIP030-model.png')
+```
+
+### Extracting Single Cell Data
+
+Identify terminal nodes (i.e. cell populations) in the workflow which were, by convention, prefixed with either ```CD4+``` or ```CD8+``` and then use those nodes to extract single-cell data into a data frame (useful for custom visualizations).
+
+```{r flow-cell-metadata, results='asis'}
```

Comment:
The results printed by this chunk are pretty ugly and my brain parsed them as debug spew. Maybe format a little? One idea would be a 2x2 table with cell types, [[list of cell types]] as the top row and cell states, [[list of cell states]] as the bottom row. I'm open to anything that looks a little better.

Reply:
Agreed that it's gross -- I'll find something prettier